### PR TITLE
[UI] [Log viewer] Prevent the infinite loading for logs fetch

### PR DESCRIPTION
**Pull request by chaowanggg**
## Why are these changes needed?
Currently, if the number of lines of logs is huge, the rendering process fails, leading to infinite loading.

To address this issue, we are implementing a maximum limit for the number of lines displayed in the LogViewer.

The LogViewer's maximum limit for lines is currently set to 932,068. However, to ensure an optimal balance between providing sufficient lines of code for debugging and maintaining acceptable performance, we have decided to set the maximum limit to 50k lines.

Please refer to the following image for more information:

image

The following is the recording describes I created 100k lines of logs and we only show 50k

ServeReplica:
[Screen Recording 2023-06-21 at 1.41.07 PM 2.zip](https://github.com/ray-project/ray/files/11825977/Screen.Recording.2023-06-21.at.1.41.07.PM.2.zip)

JobDetail:
image

ActorDetail:
image

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

**Comment by scottsun94** [comment]
Thanks! I have two questions:

1. Does it show the latest 50K lines?
2. Can we add a truncation warning at the beginning of the log?

Screenshot 2023-06-22 at 9 28 35 AM

**Comment by rkooo567** [comment]
Feel like the right solution is to just support pagination (cc @rickyyx). Maybe it is fine for a short term solution... 

Also does it mean if there are more than 50K lines, user cannot access the logs after that via dashboard?

**Comment by alanwguo** [comment]
> Feel like the right solution is to just support pagination (cc @rickyyx). Maybe it is fine for a short term solution...
> 
> Also does it mean if there are more than 50K lines, user cannot access the logs after that via dashboard?

This is a short term fix. we can do pagination in the future. But this needs to be fixed by 2.6 because it makes the log viewer un-usable if we don't limit it.

Even after the short term fix, they can still download logs to get the full logs

File: dashboard/client/src/components/LogView/LogVirtualView.tsx [hunk 1]
```diff
@@ -206,7 +208,7 @@ const LogVirtualView: React.FC<LogVirtualViewProps> = ({
         ...style,
       }}
       itemSize={fontSize + 6}
-      itemCount={total}
+      itemCount={total > maxLines ? maxLines : total}
```
Comment (alanwguo) [inline dashboard/client/src/components/LogView/LogVirtualView.tsx:211]: No, we need to limit the amount of data we fetch from the API, not just limit the number of lines we render

Comment (alanwguo) [reply dashboard/client/src/components/LogView/LogVirtualView.tsx:211]: Remember, i showed you that all the log fetching endpoints accept a `lines=x` parameter to limit the number of lines.

**Comment by rkooo567** [comment]
is it possible to add some truncation warning + ask for log download in this case? I think this is also the same UX as buildkite, so it is probably okay even for the long term fix (if we cannot find the bandwidth for pagination impl)

File: dashboard/client/src/components/LogView/LogVirtualView.tsx (hunk 1 above)
Comment (chaowanggg) [reply dashboard/client/src/components/LogView/LogVirtualView.tsx:211]: Sounds great, thanks

**Comment by chaowanggg** [comment]
> Thanks! I have two questions:
> 
> 1. Does it show the latest 50K lines?
> 2. Can we add a truncation warning at the beginning of the log?
> 
> Screenshot 2023-06-22 at 9 28 35 AM

1. Detect trancation
2. Detect empty logs

**Comment by chaowanggg** [comment]
Screenshot 2023-06-29 at 5 38 37 PM

Updated warning info 
@scottsun94 

One edge case is the user actually have 50k logs and not tailed, we will still show the warning info here

**Review by alanwguo — APPROVED** [review]

**Comment by scottsun94** [comment]
LGTM

**Pull request merged**